src/Formatter/PlatesFormatter.php

```diff
@@ -26,27 +25,42 @@ public function __construct(Engine $engine)
      */
     public function body(PayloadInterface $payload)
     {
-        $template = $this->template($payload);
-        $template = $this->engine->make($template);
-        return $this->render($template, $payload);
+        return $this->render($payload);
     }
 
     /**
      * @param PayloadInterface $payload
+     *
      * @return string
      */
+    protected function render(PayloadInterface $payload)
+    {
+        $template = $this->template($payload);
+        $output = $this->output($payload);
+
+        return $this->engine->render($template, $output);
+    }
+
+    /**
+     * @param PayloadInterface $payload
+     *
+     * @return string Template name
+     */
     protected function template(PayloadInterface $payload)
     {
         return $payload->getOutput()['template'];
     }
 
     /**
-     * @param Template $template
      * @param PayloadInterface $payload
-     * @return string
+     *
+     * @return array $output
      */
-    protected function render(Template $template, PayloadInterface $payload)
+    protected function output(PayloadInterface $payload)
     {
-        return $template->render($payload->getOutput());
+        $output = $payload->getOutput();
+        unset($output['template']);
```

Comment:
I don't know of a situation where you would define template names for JSON formatting. Perhaps what this really speaks to is a need for better way to pass the template name in the payload?